extensions/wikia/AchievementsII/services/AchAwardingService.class.php

```diff
@@ -124,22 +124,22 @@ public function processSharing( $articleID, $sharerID, $IP ) {
 		 *  ]
 		 * ]
 		 */
-		if (empty( $this->mCounters[BADGE_SHARING] ) ) {
-			$this->mCounters[BADGE_SHARING] = array( 'ips' => array(), 'article_ids' => array() );
+		if( empty( $this->mCounters[ BADGE_SHARING ] ) ) {
+			$this->mCounters[ BADGE_SHARING ] = array( 'ips' => array(), 'article_ids' => array() );
 		}
 
-		if ( isset( $this->mCounters[BADGE_SHARING]['article_ids'][$articleID] ) ) {
+		if( isset( $this->mCounters[ BADGE_SHARING ][ 'article_ids' ][ $articleID ] ) ) {
 			// called from special page to which shared link lead to
-			if (!in_array( $IP, $this->mCounters[BADGE_SHARING]['ips'] ) ) {
-				if ( !in_array( $IP, $this->mCounters[BADGE_SHARING]['article_ids'][$articleID] ) ) {
-					$this->mCounters[BADGE_SHARING]['article_ids'][$articleID][] = $IP;
+			if( !in_array( $IP, $this->mCounters[ BADGE_SHARING ][ 'ips' ] ) ) {
+				if( !in_array( $IP, $this->mCounters[ BADGE_SHARING ][ 'article_ids' ][ $articleID ] ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)